src/codegen/llvm.zig

```diff
@@ -1459,7 +1459,8 @@ pub const Object = struct {
                     .signed => DW.ATE.signed,
                     .unsigned => DW.ATE.unsigned,
                 };
-                const di_type = dib.createBasicType(name, info.bits, dwarf_encoding);
+                const di_bits = ty.abiSize(target) * 8; // lldb cannot handle non-byte sized types
```

Comment:
We should probably file an upstream issue for this. If the DWARF spec allows for non-byte-sized types, `lldb` should fix it upstream.

Reply:
Ironically, I so far relied on `DW_AT_byte_size` for self-hosted native backends. For example https://github.com/ziglang/zig/blob/c679fdee7fa4999d79ba741aeefa1cf7e1756f0b/src/link/Dwarf.zig#L195

Reply:
> We should probably file an upstream issue for this. If the DWARF spec allows for non-byte-sized types, `lldb` should fix it upstream.

They work in gdb so it does seem like a bug in lldb.

> Is there a way to tell LLVM to use that attribute instead of `DW_AT_bit_size`? It might be safer this way. However, if in addition we wanted to capture the fact that type takes a non-byte-size size, we could use a combination of `DW_AT_byte_size`, `DW_AT_bit_size` and `DW_AT_data_bit_offset`.

It doesn't look possible.
```c++
void DwarfUnit::constructTypeDIE(DIE &Buffer, const DIBasicType *BTy) {
...
  uint64_t Size = BTy->getSizeInBits() >> 3;
  addUInt(Buffer, dwarf::DW_AT_byte_size, None, Size);
```

Reply:
> > We should probably file an upstream issue for this. If the DWARF spec allows for non-byte-sized types, `lldb` should fix it upstream.
> 
> They work in gdb so it does seem like a bug in lldb.

Fair point.

> > Is there a way to tell LLVM to use that attribute instead of `DW_AT_bit_size`? It might be safer this way. However, if in addition we wanted to capture the fact that type takes a non-byte-size size, we could use a combination of `DW_AT_byte_size`, `DW_AT_bit_size` and `DW_AT_data_bit_offset`.
> 
> It doesn't look possible.
> 
> ```c++
> void DwarfUnit::constructTypeDIE(DIE &Buffer, const DIBasicType *BTy) {
> ...
>   uint64_t Size = BTy->getSizeInBits() >> 3;
>   addUInt(Buffer, dwarf::DW_AT_byte_size, None, Size);
> ```

Argh, that seems limiting...